web/src/main/webapp/js/main-template.js

```diff
@@ -554,18 +554,40 @@ function routeLatLng(request, doQuery) {
             mapLayer.addDataToRoutingLayer(geojsonFeature);
             var oneTab = $("<div class='route_result_tab'>");
             routeResultsDiv.append(oneTab);
-            tabHeader.click(createClickHandler(geoJsons, pathIndex, tabHeader, oneTab, request.hasElevation()));
+            tabHeader.click({useMiles: request.useMiles}, createClickHandler(geoJsons, pathIndex, tabHeader, oneTab, request.hasElevation()));
```

Comment:
I now think we actually do not need it and can pass use_miles directly to createClickHandler as you say. I think I tried this before and the elevation inset did not get updated when the unit buttons were clicked, but I must have made something wrong then. I got it working now and of course this is the better solution.